app/gui/src/presenter/graph/state.rs

```diff
@@ -117,16 +118,22 @@ impl Nodes {
         view_id: ViewNodeId,
         ast_id: AstNodeId,
     ) -> &mut Node {
-        let mut displayed =
-            Self::get_mut_or_create_static(&mut self.nodes, &mut self.nodes_without_view, ast_id);
-        if let Some(old_view) = displayed.view_id {
-            self.ast_node_by_view_id.remove(&old_view);
+        let mut displayed = self.nodes.entry(ast_id).or_default();
+        let old_ast_to_remove = if let Some(old_view) = displayed.view_id {
+            self.ast_node_by_view_id.remove(&old_view)
         } else {
-            self.nodes_without_view.remove_item(&ast_id);
```

Comment:
I guess you're right.

My initial thought process was the following:
- We had this line in the previous version of the function because of `get_mut_or_create_static`, which adds a new node to the `nodes_without_view` list.
- We can safely skip this step because we don't use `get_mut_or_create_static` anymore.

Now I see that there are possible situations when we create the node by other means and only then connect it to the view using this method. I didn't find any such code path in our current implementation, but I guess this can change in the future. After all, removing the item is a safe action here. I added it back in the commit 939549b0dd7d3a671402dd8d0ea02dbb0e94a930